src/mdlComponentHandler.js

```diff
@@ -402,6 +402,27 @@ componentHandler = (function() {
     }
   }
 
+  /**
+  * Remove the element from the internal createdComponents_ array.
+  *
+  * @param {!Element} element
+  */
+  function removeComponentInternal(element) {
+    var foundComponents = [];
+    for (var n = 0; n < createdComponents_.length; n++) {
+      var component = createdComponents_[n];
+      if (component.element_ === element) {
+        foundComponents.push(component);
+      }
+    }
+    for (var i = 0, length = foundComponents.length; i < length; i++) {
+      createdComponents_.splice(createdComponents_.indexOf(foundComponents[i]), 1);
+    }
+    var ev = document.createEvent('Events');
+    ev.initEvent('mdl-componentremoved', true, false);
```

Comment:
Yea, someone could toss in any random node and get an event when nothing was actually done. Good thinking.